src/integration/java/org/springframework/data/couchbase/TestContainerResource.java

```diff
@@ -0,0 +1,66 @@
+package org.springframework.data.couchbase;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.rules.ExternalResource;
+import org.testcontainers.containers.FixedHostPortGenericContainer;
+
+/**
+ * Testcontainers as external resource. It is recommended to use it as ClassRule.
+ * It also does the internal reference counting, in case if the getResource is called again.
+ *
+ */
+public class TestContainerResource extends ExternalResource {
+
+	private static FixedHostPortGenericContainer couchbaseContainer = null;
+	private static AtomicInteger referenceCount = new AtomicInteger();
+	private static TestContainerResource currentInstance;
+	private static String serverVersion;
+
+	public static TestContainerResource getResource() {
+		if (currentInstance == null) {
+			currentInstance = new TestContainerResource();
+			try {
+				Properties properties = new Properties();
+				properties.load(Thread.currentThread().getContextClassLoader().getResourceAsStream("server.properties"));
+				serverVersion = properties.getProperty("server.version");
+				if(properties.getProperty("server.resource").compareTo("container") != 0) {
+					return null;
+				}
+			} catch (IOException ex) {
+				serverVersion = "5.0.1";
+			}
+			couchbaseContainer = new FixedHostPortGenericContainer("couchbase:" + serverVersion)
```

Comment:
as long as we start only a single container it should be fine... I came up with a way to map dynamically but this might be too much for here until we converge? is it only ever starting one?

Reply:
Mapping dynamically is possible, just a bit more hassle, anyways our ports dont change, it is a one time thing.